client/src/Features/MicrosoftLearn/MicrosoftLearnFilterComponentUtils.tsx

```diff
@@ -0,0 +1,151 @@
+import { useObserver } from 'mobx-react-lite';
+import React from 'react';
+import {
+  getProductsToDisplay,
+  getTypesToDisplay,
+  getLevelsToDisplay,
+  getRolesToDisplay
+} from './MicrosoftLearnFilterUtils';
+import { FilterType } from '../../Models/Learn/FilterType.model';
+import { MicrosoftLearnFilterComponent } from './MicrosoftLearnFilterComponent';
+import { FontWeights } from '@fluentui/react';
+import { IThemeOnlyProps, SimpleComponentStyles } from '../../Core/Utils/FluentUI/typings.fluent-ui';
+import { themedClassNames } from '../../Core/Utils/FluentUI';
+
+export type FilterComponentStyles = SimpleComponentStyles<
+  'root' | 'title' | 'search' | 'optionsList' | 'subOptionsList' | 'filterItem'
+>;
+
+type FilterComponentTypes = FilterType.products | FilterType.roles | FilterType.levels | FilterType.types;
+
+export const FilterComponent = (props: { type: FilterComponentTypes; name: string }) => {
+  const { filterStore, catalog } = useStore('microsoftLearnStore');
+  return useObserver(() => {
+    
+    const getDisplayOptions = () => {
+      switch (props.type) {
+        case FilterType.products:
+          return getProductsToDisplay(filterStore.displayFilter[props.type], catalog?.products);
+        case FilterType.roles:
+          return getRolesToDisplay(filterStore.displayFilter[props.type], catalog?.roles);
+        case FilterType.levels:
+          return getLevelsToDisplay(filterStore.displayFilter[props.type], catalog?.levels);
+        case FilterType.types:
+          return getTypesToDisplay(filterStore.displayFilter[props.type]);
+      }
+    };
+
+    const displayOptions = getDisplayOptions();
+    if (props.type === FilterType.products) {
+      return (
+        <>
+          <MicrosoftLearnFilterComponent
+            styles={themedClassNames(FilterComponentStyles)}
+            filterType={props.type}
+            filterName={props.name}
+            filterOption={displayOptions}
+            search={true}
+            mainItemClickHandler={event => {
+              let target = event?.target as HTMLInputElement;
+              let value = target.getAttribute('aria-describedby');
+              if (value) {
+                let subItems: string[] = [...catalog?.products.values()]
+                  .filter(product => product.parentId && product.parentId === value)
+                  .map(product => product.id);
+                target.checked
+                  ? filterStore.addFilter(props.type, [...subItems, value])
+                  : filterStore.removeFilter(props.type, [...subItems, value]);
+              }
+            }}
+            subItemClickHandler={event => {
+              let target = event?.target as HTMLInputElement;
+              let type = FilterType.products;
+              let value = target.getAttribute('aria-describedby');
+              if (target.checked && value) {
+                filterStore.addFilter(type, [value]);
+              } else if (!target.checked && value) {
```

Comment:
can we use `else` instead of `else if` here and in line 93?

Reply:
The `value` for the `target` could be `null`! An alternative could be to wrap the `if-else` block inside another `if` check for the `value`, but this seemed cleaner than that, no?